Detectors/MUON/MCH/Workflow/src/DataDecoder.cxx

```diff
@@ -0,0 +1,602 @@
+                                                                    stopTime, false, solarId, dsAddr, chAddr});
+}
+
+void Merger::mergeDigits(int feeId)
+{
+  if (feeId < 0 || feeId > MCH_MERGER_FEEID_MAX) {
+    return;
+  }
+
+  mergers[feeId].mergeDigits();
+}
+
+//=======================
+// Data decoder
+void DataDecoder::decodeBuffer(gsl::span<const std::byte> page)
+{
+  size_t ndigits{0};
+
+  uint8_t isStopRDH = 0;
+  uint32_t orbit;
+  uint32_t feeId;
+  uint32_t linkId;
+
+  const auto storeDigit = [&](const Digit& d) {
+    mOutputDigits.emplace_back(d);
+    if (mPrint) {
+      std::cout << "[storeDigit]: digit stored, mOutputDigits size: " << mOutputDigits.size() << std::endl;
+    }
+  };
+
+  auto channelHandler = [&](DsElecId dsElecId, uint8_t channel, o2::mch::raw::SampaCluster sc) {
+    if (mChannelHandler) {
+      mChannelHandler(dsElecId, channel, sc);
+    }
+
+    if (mDs2manu) {
+      channel = ds2manu(int(channel));
+    }
+
+    double digitadc(0);
+    for (auto& s : sc.samples) {
+      digitadc += s;
+    }
+
+    int deId{-1};
+    int dsIddet{-1};
+    if (auto opt = mElec2Det(dsElecId); opt.has_value()) {
+      DsDetId dsDetId = opt.value();
+      dsIddet = dsDetId.dsId();
+      deId = dsDetId.deId();
+    }
+
+    int padId = -1;
+    const Segmentation& segment = segmentation(deId);
+    if ((&segment) == nullptr) {
+      return;
+    }
+
+    padId = segment.findPadByFEE(dsIddet, int(channel));
+    if (mPrint) {
+      auto s = asString(dsElecId);
+      auto ch = fmt::format("{}-CH{:02d}", s, channel);
+      std::cout << ch << "  "
+                << fmt::format("PAD ({:04d} {:04d} {:04d})\tADC {:5.0f}  TIME ({} {} {:02d})  SIZE {}  END {}",
+                               deId, dsIddet, padId, digitadc, orbit, sc.bunchCrossing, sc.sampaTime, sc.nofSamples(), (sc.sampaTime + sc.nofSamples() - 1))
+                << (((sc.sampaTime + sc.nofSamples() - 1) >= 98) ? " *" : "") << std::endl;
+    }
+
+    // skip channels not associated to any pad
+    if (padId < 0) {
+      return;
+    }
+
+    Digit::Time time;
+    time.sampaTime = sc.sampaTime;
+    time.bunchCrossing = sc.bunchCrossing;
+    time.orbit = orbit;
+
+    Digit::Time stopTime;
+    stopTime.sampaTime = sc.sampaTime + sc.nofSamples() - 1;
+    stopTime.bunchCrossing = sc.bunchCrossing;
+    stopTime.orbit = orbit;
+
+    mMerger->addDigit(feeId, static_cast<int>(dsElecId.solarId()), static_cast<int>(dsElecId.elinkId()), static_cast<int>(channel),
+                      deId, padId, digitadc, time, stopTime);
+    ++ndigits;
+  };
+
+  const auto patchPage = [&](gsl::span<const std::byte> rdhBuffer) {
+    auto rdhAnyPtr = reinterpret_cast<o2::header::RDHAny*>(const_cast<std::byte*>(&rdhBuffer[0]));
+    if (mRdhHandler) {
+      mRdhHandler(rdhAnyPtr);
+    }
+    auto rdhPtr = reinterpret_cast<o2::header::RAWDataHeaderV4*>(const_cast<std::byte*>(&rdhBuffer[0]));
```

Comment:
should be replaced by 
```
const void* rdhP = reinterpret_cast<const void*>(rdhBuffer.data());
```
to get a RDH-version agnostic code

Reply:
Actually, what is the point of having ``o2::header::RDHAny* rdhAnyPtr`` a few lines above and then the same pointer as ``RAWDataHeaderV4`` (or even ``void`` as @aphecetche suggest?).
It seems it is never used, as well as its dereferenced version on L464.
In general, I would suggest to use the reference on RDHAny, i.e.
````
using RDH = o2::header::RDHAny;
....
auto& rdhAny = *reinterpret_cast<RDH*>(const_cast<std::byte*>(&rdhBuffer[0]));
````
This will provide the same RDHUtils functionality as using the pointer (all methods are interfaced to both pointers and refs),  but has an advantage that once we will settle on the final header, we can alias RDH to this header. Then, the tiny overhead we can have in RDHUtils due to the version check will be fully avoided.

Reply:
I have adopted the suggestion for @shahor02, hopefully in all relevant places.